core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/visualization/barChart/BarChartOpDesc.scala

```diff
@@ -1,140 +1,132 @@
+    s"""
+       |        table = table.dropna() #remove missing values
+       |""".stripMargin
   }
 
-  private def getGroupByKeysSchema(schemas: Array[Schema]): Schema = {
-    val groupByKeys = List(this.nameColumn)
-    Schema
-      .newBuilder()
-      .add(groupByKeys.map(key => schemas(0).getAttribute(key)).toArray: _*)
-      .build()
-  }
+  override def numWorkers() = 1
 
-  private def getFinalAggValueSchema: Schema = {
-    if (noDataCol) {
-      Schema
-        .newBuilder()
-        .add(resultAttributeNames.head, AttributeType.INTEGER)
-        .build()
-    } else {
-      Schema
-        .newBuilder()
-        .add(resultAttributeNames.map(key => new Attribute(key, AttributeType.DOUBLE)).toArray: _*)
-        .build()
-    }
+  def createPlotlyFigure(): String = {
+    assert(value.nonEmpty)
+    assert(fields.nonEmpty)
+    s"""
+       |fig = px.bar(table, y= '$fields', x='$value', orientation = 'h')
+       |""".stripMargin
   }
 
-  override def operatorExecutor(operatorSchemaInfo: OperatorSchemaInfo): OpExecConfig = {
-    throw new UnsupportedOperationException("multi layer operators use operatorExecutorMultiLayer")
+  override def generatePythonCode(operatorSchemaInfo: OperatorSchemaInfo): String = {
+    var truthy = ""
+    if (Orientation) truthy = "True"
+    val final_code = s"""
+                        |from pytexera import *
+                        |
+                        |import plotly.express as px
+                        |import pandas as pd
+                        |import plotly.graph_objects as go
+                        |import plotly.io
+                        |import json
+                        |import pickle
+                        |import plotly
+                        |
+                        |class ProcessTableOperator(UDFTableOperator):
+                        |
+                        |    @overrides
+                        |    def process_table(self, table: Table, port: int) -> Iterator[Optional[TableLike]]:
+                        |        ${manipulateTable()}
+                        |        if not table.empty and '$fields' != '$value':
+                        |           if ($truthy):
+                        |              fig = go.Figure(px.bar(table, y='$fields', x='$value', orientation = 'h', title='$title'))
+                        |           else:
+                        |              fig = go.Figure(px.bar(table, y='$value', x='$fields', title='$title'))
+                        |           html = plotly.io.to_html(fig, include_plotlyjs = 'cdn', auto_play = False)
+                        |           # use latest plotly lib in html
+                        |           #html = html.replace('https://cdn.plot.ly/plotly-2.3.1.min.js', 'https://cdn.plot.ly/plotly-2.18.2.min.js')
+                        |           yield {'html-content':html}
+                        |        else:
+                        |           html = '''<h1>Bar Chart is not available.</h1>
```

Comment:
We want to make the error reporting more clear and meaningful to the user, you can check my PR to see what I did.